plugins/epochstorage/plugin.go

```diff
@@ -0,0 +1,474 @@
+		committedEpochsMutex.Unlock()
+		return
+	}
+	committableEpochs = committableEpochs[len(committableEpochs)-maxEpochContentsToKeep:]
+	committedEpochsMutex.Unlock()
+}
+
+func GetCommittedEpochs() (ecRecords map[epoch.Index]*epoch.ECRecord) {
+	ecRecords = make(map[epoch.Index]*epoch.ECRecord, 0)
+
+	committedEpochsMutex.RLock()
+	for _, record := range committableEpochs {
+		ecRecords[record.EI()] = record
+	}
+	committedEpochsMutex.RUnlock()
+
+	return
+}
+
+func GetPendingBranchCount() map[epoch.Index]uint64 {
+	return deps.NotarizationMgr.PendingConflictsCountAll()
+}
+
+func GetEpochMessages(ei epoch.Index) ([]tangle.MessageID, error) {
+	stores, err := getEpochContentStorage(ei)
+	if err != nil {
+		return []tangle.MessageID{}, err
+	}
+
+	var msgIDs []tangle.MessageID
+	stores.messageIDs.IterateKeys(kvstore.EmptyPrefix, func(key kvstore.Key) bool {
+		var msgID tangle.MessageID
+		if _, err := msgID.Decode(key); err != nil {
+			panic("MessageID could not be parsed!")
+		}
+		msgIDs = append(msgIDs, msgID)
+		return true
+	})
+
+	return msgIDs, nil
+}
+
+func GetEpochTransactions(ei epoch.Index) ([]utxo.TransactionID, error) {
+	stores, err := getEpochContentStorage(ei)
+	if err != nil {
+		return []utxo.TransactionID{}, err
+	}
+
+	var txIDs []utxo.TransactionID
+	stores.transactionIDs.IterateKeys(kvstore.EmptyPrefix, func(key kvstore.Key) bool {
+		var txID utxo.TransactionID
+		if _, err := txID.Decode(key); err != nil {
+			panic("TransactionID could not be parsed!")
+		}
+		txIDs = append(txIDs, txID)
+		return true
+	})
+
+	return txIDs, nil
+}
+
+func GetEpochUTXOs(ei epoch.Index) (spent, created []utxo.OutputID, err error) {
+	stores, err := getEpochContentStorage(ei)
+	if err != nil {
+		return []utxo.OutputID{}, []utxo.OutputID{}, err
+	}
+
+	stores.createdOutputs.IterateKeys(kvstore.EmptyPrefix, func(key kvstore.Key) bool {
+		outputID, err := outputIDFromBytes(key)
+		if err != nil {
+			panic(err)
+		}
+		created = append(created, outputID)
+
+		return true
+	})
+
+	stores.spentOutputs.IterateKeys(kvstore.EmptyPrefix, func(key kvstore.Key) bool {
+		outputID, err := outputIDFromBytes(key)
+		if err != nil {
+			panic(err)
+		}
+		spent = append(spent, outputID)
+
+		return true
+	})
+
+	return spent, created, nil
+}
+
+func GetEpochVotersWeight(ei epoch.Index) (weights map[epoch.ECR]map[identity.ID]float64) {
+	epochVotersWeightMutex.RLock()
+	defer epochVotersWeightMutex.RUnlock()
+	if _, ok := epochVotersWeight[ei]; !ok {
+		return
+	}
+
+	duplicate := make(map[epoch.ECR]map[identity.ID]float64, len(epochVotersWeight[ei]))
+	for k, v := range epochVotersWeight[ei] {
+		subDuplicate := make(map[identity.ID]float64, len(v))
+		for subK, subV := range v {
+			subDuplicate[subK] = subV
+		}
+		duplicate[k] = subDuplicate
+	}
+	return duplicate
+}
+
+func getEpochContentStorage(ei epoch.Index) (*epochContentStorages, error) {
+	if ei != epoch.Index(math.MaxInt64) && ei < minEpochIndex {
```

Comment:
It shouldn't be necessary to check for the first condition, the epoch.Index you receive is already an uint64 under the hood, and therefore its capacity is already bounded.